addons/test_mail/tests/test_mail_composer.py

```diff
@@ -453,6 +453,37 @@ def test_mail_composer_rights_attachments(self):
             sorted(self.test_record.message_ids[0].attachment_ids.mapped('name')),
             sorted(template_1_attachment_name))
 
+    @users('employee')
+    def test_mail_composer_template_switching(self):
+        """ Ensure that the current user's identity serves as the sender,
+        particularly when transitioning from a template with a designated sender to one lacking such specifications.
+        Moreover, we verify that switching to a template lacking any subject maintains the existing subject intact.
```

Comment:
Note that I would test reply_to reset, as it is part of this patch :)

Reply:
Good catch! I checked again on `reply_to` behavior and it seem correct without my fix, which isn't doing anything for "reply_to" as it doesn't have any specific logic in the [`MailComposer.default_get`](https://github.com/odoo/odoo/blob/a161e7a2dd46ee1b25cc1b64161fc43864b007cb/addons/mail/wizard/mail_compose_message.py#L43) nor does it have any default value. 

-> simplifying to email_from only 

![image](https://github.com/odoo/odoo/assets/33809926/0b16f16a-7979-4b3b-8783-7bcdb3356d1d)